### DISCUSSION TITLE
Solar panel counting using SAM

### DISCUSSION THREAD

**Discussion by shamukshi** [opengeos/segment-geospatial#193] [Q&A] [answered]
For "solar panel counting from UAV image using SAM"....
which approach is better >>> 1. 'objection detection followed by segmentation' or 2. 'segmentation followed by classification'

**Comment by giswqs** [comment, marked as answer]
See this example: https://samgeo.gishub.org/examples/swimming_pools/

**Comment by jamescrowley** [comment]
@shamukshi did you have any success with this? I've been experimenting with the model using the swimming pool example as a base, but not found an effective combination of parameters and prompts yet - most of the results are way off.

**Comment by shamukshi** [comment]
hi...
so my target was solar panel.....so when i use "solar panels" as
prompt...it gave me solar panel...
When I used "glass" as a prompt....it still gave me solar panels...
a bit same for other prompts too...
I guess that AI prompt technology is still in the works.

So I just went with the traditional method...I used SAM to create segments
(unclassified segments) from my UAV images...
Trained a classifier to classify those SAM segments as solar panel and
non-solar panel
